axon-reactor/src/main/java/org/axonframework/extensions/reactor/queryhandling/gateway/DefaultReactorQueryGateway.java

```diff
@@ -200,12 +201,11 @@ public Registration registerResultHandlerInterceptor(
     @SuppressWarnings("unchecked")
     private <R> Flux<R> getPayload(Flux<ResultMessage<?>> resultMessageFlux) {
         return resultMessageFlux
-                .flatMap(this::mapResult)
                 .filter(r -> Objects.nonNull(r.getPayload()))
                 .map(it -> (R) it.getPayload());
     }
 
-    private Flux<? extends ResultMessage<?>> mapResult(ResultMessage<?> response) {
+    private Flux<? extends ResultMessage<?>> mapExceptionalResult(ResultMessage<?> response) {
```

Comment:
Nit: I'd rename the parameter to `result` instead of `response`, to more closely align with the type.